src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,115 @@
+* */
 public class StringManipulation implements StringManipulationInterface {
 
+    String string; // holding string input
+
     @Override
-    public String getString() {
-        return null;
+    public String getString() { // method to return string
+        return string;
     }
 
     @Override
-    public void setString(String string) {
+    public void setString(String string) { // method to set the string
+        this.string = string;
     }
 
+    // this method will count the # of words in the string
     @Override
     public int count() {
-        return 0;
+        // throw null pointer exception if string is null
+        if (string == null) {
+            throw new NullPointerException("String is null");
+        }
+        // if string is empty throw an exception
+        if (string.isEmpty()) {
+            throw new IndexOutOfBoundsException("string is empty!");
+        }
+        // creating an array for the string
+        String[] words = string.split("\\s+");
+        // returning length of words(string count)
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        // if n < 0, throw an error
+        if (n <= 0) {
+            throw new IllegalArgumentException("'n' must be greater than zero.");
+        }
+        // if n > length, throw an error
+        if (n > string.length()) {
+            throw new IndexOutOfBoundsException("'n' is greater than the string length.");
+        }
+        // catching null errors
+        if (string == null) {
+            throw new NullPointerException("String is null!");
+        }
+        // string builder to manipulate
+        StringBuilder result = new StringBuilder();
+        // setting count to 1
+        int count = 1;
+        // looping through the string
+        for (int i = 0; i < string.length(); i++) {
+            if (count % n != 0) {
+                result.append(string.charAt(i));// append char to the result string
+            } else if (maintainSpacing) {
+                result.append(' ');// append a space to maintain spacing
+            }
+            count++;// increment count
+        }
+
+        return result.toString();// return modified string
     }
 
     @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        // catching null string
+        if (string == null) {
+            throw new NullPointerException("String cannot be null!");
+        }
+        String[] words = string.trim().split("\\s+");// splitting the string into words
+        // throwing an error if out of bounds
+        if (startWord <= 0 || endWord <= 0 || startWord > endWord || endWord > words.length) {
+            throw new IllegalArgumentException("Invalid values for startWord and endWord");
+        }
+        // catching end word larger than length of string
+        if (endWord > words.length) {
```

Comment:
unreachable code, as if the condition is true, it will throw IllegalArgumentException in line 80

Reply:
removed